python_multipart/multipart.py

```diff
@@ -1652,7 +1653,9 @@ def on_header_value(data: bytes, start: int, end: int) -> None:
                 header_value.append(data[start:end])
 
             def on_header_end() -> None:
-                headers[b"".join(header_name)] = b"".join(header_value)
+                # Convert header name to title case.
+                header_name_tc = b"".join(header_name).decode().title()
+                headers[header_name_tc] = b"".join(header_value)
```

Comment:
Shouldn't we actually lower case instead of `title`?

Reply:
Either would work, there's a trade off. 

Lower case will require more of the code and tests to change as they currently expect title case. It also impacts the outside interface, where we say "The only required header is Content-Type." and don't say anything about the case-sensitivity of the input dict. I don't want to have to re-ingest the input header dict to ensure it's case insensitive. Switching to using only the content-type header value here would help, but entirely breaks the external API.